core/src/transport/upgrade.rs

```diff
@@ -80,6 +80,7 @@ where
     }
 }
 
+/// Future to avoid boxing.
```

Comment:
We shouldn't comment on the intent of the code. This information is totally useless to the user. This would be similar to saying "this is a function to reduce code duplication". It should be something like `Future that upgrades the dialing side to use an upgrade`. Same for the others.